### PR TITLE
Use test app for Sessions e2e test

### PR DESCRIPTION
**Pull request by mrober**

### CODE REVIEW

File: .github/workflows/sessions-e2e.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ env:
   SESSIONS_E2E_GOOGLE_SERVICES: ${{ secrets.SESSIONS_E2E_GOOGLE_SERVICES }}
 
 jobs:
-  build:
+  test:
```
[the remark this answers is not in the document]
Comment (mrober) [reply .github/workflows/sessions-e2e.yml:15]: The `build` label was left over from the template I used to start making this workflow in an earlier PR. This is actually a test. So I think this label makes more sense.

File: firebase-sessions/test-app/test-app.gradle.kts [hunk 2]
```diff
@@ -46,8 +51,15 @@ dependencies {
   implementation("androidx.constraintlayout:constraintlayout:2.1.4")
   implementation("androidx.core:core-ktx:1.9.0")
   implementation("com.google.android.material:material:1.8.0")
+
+  androidTestImplementation("com.google.firebase:firebase-common-ktx:20.3.2")
+  androidTestImplementation(libs.androidx.test.junit)
+  androidTestImplementation(libs.androidx.test.runner)
+  androidTestImplementation(libs.truth)
```
[the remark this answers is not in the document]
Comment (mrober) [reply firebase-sessions/test-app/test-app.gradle.kts:58]: I prefer using Guava's Truth than JUnit's regular asserts. Truth reads much better, and other Firebase SDKs use it in their tests too.

File: firebase-sessions/test-app/src/androidTest/kotlin/com/google/firebase/testing/sessions/FirebaseSessionsTest.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  @Test
+  fun initializeSession() {
+    // Force the Firebase Sessions SDK to initialize.
+    assertThat(FirebaseSessions.instance.greeting()).isEqualTo("Matt says hi!")
+
+    // Wait for the session start event to send.
+    // TODO(mrober): Setup logger we can access from tests.
+    Thread.sleep(40_000)
```
[the remark this answers is not in the document]
Comment (mrober) [reply firebase-sessions/test-app/src/androidTest/kotlin/com/google/firebase/testing/sessions/FirebaseSessionsTest.kt]: Done

File: firebase-sessions/test-app/test-app.gradle.kts (hunk 2 above)
[the remark this answers is not in the document]
Comment (visumickey) [reply firebase-sessions/test-app/test-app.gradle.kts:58]: My comment was in the context of why we use that for the test app. For SDK's it makes sense to me. Do we have tests for the test app and we are using Guava's truth library for that?

**Review by visumickey — APPROVED** [review]

File: firebase-sessions/test-app/test-app.gradle.kts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrober) [reply firebase-sessions/test-app/test-app.gradle.kts:58]: Oh is there any reason we should avoid it?

Right now we do 1 `Truth.assertThat(...)` on the placeholder api. That is how the test verifies that Sessions and everything got initialized correctly in the test app for now. We could make a simpler assert like that the instance is not null? But after my TODO is resolved, I expect to add different asserts on the logs. E.g. assert that the session was successfully sent to firelog.

**Pull request merged**